tests/test_node.py

```diff
@@ -1,12 +1,34 @@
 import pytest
 
-from travertino.declaration import BaseStyle
+from tests.utils import mock_attr, prep_style_class
+from travertino.declaration import BaseStyle, Choices, validated_property
 from travertino.layout import BaseBox, Viewport
 from travertino.node import Node
 from travertino.size import BaseIntrinsicSize
 
 
+@prep_style_class
+@mock_attr("reapply")
 class Style(BaseStyle):
+    int_prop: int = validated_property(Choices(integer=True))
+
+    class IntrinsicSize(BaseIntrinsicSize):
+        pass
+
+    class Box(BaseBox):
+        pass
+
+    def layout(self, root, viewport):
+        # A simple layout scheme that allocates twice the viewport size.
+        root.layout.content_width = viewport.width * 2
+        root.layout.content_height = viewport.height * 2
+
+
+@prep_style_class
+class BrokenStyle(BaseStyle):
+    def reapply(self):
+        raise AttributeError
```

Comment:
This should be an instance, not the class.

Reply:
I don't think that will work. When a style is assigned to the node, it's copied, so anything patched onto the original instance is lost, and you're left with a new instance of whatever class it was.

Reply:
I meant returning `raise AttributeError()` rather than `raise AttributeError`; but it looks like that's a moot point now.